core/trino-main/src/main/java/io/trino/execution/SqlTaskExecution.java

```diff
@@ -574,15 +574,22 @@ private DriverSplitRunnerFactory(DriverFactory driverFactory, boolean partitione
             this.pipelineContext = taskContext.addPipelineContext(driverFactory.getPipelineId(), driverFactory.isInputDriver(), driverFactory.isOutputDriver(), partitioned);
         }
 
-        // TODO: split this method into two: createPartitionedDriverRunner and createUnpartitionedDriverRunner.
-        // The former will take two arguments, and the latter will take one. This will simplify the signature quite a bit.
-        public DriverSplitRunner createDriverRunner(@Nullable ScheduledSplit partitionedSplit)
+        public DriverSplitRunner createPartitionedDriverRunner(ScheduledSplit partitionedSplit)
+        {
+            return createDriverRunner(requireNonNull(partitionedSplit, "partitionedSplit is null"), partitionedSplit.getSplit().getSplitWeight().getRawValue());
```

Comment:
requireNonNull is not needed since `partitionedSplit` is already dereferenced here (https://www.javacodegeeks.com/2020/07/better-nullpointerexception-messages-automatic-in-jdk-15.html)